ush/fv3gfs_downstream_nems.sh

```diff
@@ -1,294 +1,177 @@
-        cat pgb2file_${fhr3}_${iproc}_0p5  >> pgb2file_${fhr3}_0p5
-        cat pgb2file_${fhr3}_${iproc}_1p0  >> pgb2file_${fhr3}_1p0
-        if [ "$PGB1F" = 'YES' ]; then
-          cat pgbfile_${fhr3}_${iproc}_1p0   >> pgbfile_${fhr3}_1p0
-        fi
-      fi
-    elif [ $nset = 2 ]; then
-      cat pgb2bfile_${fhr3}_${iproc}_0p25 >> pgb2bfile_${fhr3}_0p25
-      if [ "$PGBS" = "YES" ]; then
-        cat pgb2bfile_${fhr3}_${iproc}_0p5 >> pgb2bfile_${fhr3}_0p5
-        cat pgb2bfile_${fhr3}_${iproc}_1p0 >> pgb2bfile_${fhr3}_1p0
-      fi
-    fi
-    export iproc=$(expr $iproc + 1)
+  err_chk
+
+  # We are in a loop over downset, save output from mpmd into nset specific output
+  cat mpmd.out  # so we capture output into the main logfile
+  mv mpmd.out "mpmd_${nset}.out"
+
+  # Concatenate grib files from each processor into a single one
+  # and clean-up as you go
+  echo "Concatenating processor specific grib2 files into a single product"
+  for (( iproc = 1 ; iproc <= nproc ; iproc++ )); do
+    for grid in "${grids[@]}"; do
+      cat "pgb2${grp}file_${fhr3}_${iproc}_${grid}" >> "pgb2${grp}file_${fhr3}_${grid}"
+      rm  "pgb2${grp}file_${fhr3}_${iproc}_${grid}"
+    done
+    # There is no further use of the processor specific tmpfile; delete it
+    rm "${tmpfile}_${iproc}"
   done
-  date
 
-  #Chuang: generate second land mask using bi-linear interpolation and append to the end
-  # if [ $nset = 1 ]; then
-  # rm -f land.grb newland.grb newnewland.grb newnewland.grb1
-  # $WGRIB2 $tmpfile -match "LAND:surface" -grib land.grb
-  ##0p25 degree
-  # $WGRIB2 land.grb -set_grib_type same -new_grid_interpolation bilinear -new_grid_winds earth -new_grid $grid0p25 newland.grb
-  # $WGRIB2 newland.grb -set_byte 4 11 218 -grib newnewland.grb
-  # cat ./newnewland.grb >> pgb2file_${fhr3}_0p25
-  # $CNVGRIB -g21 newnewland.grb newnewland.grb1
-  # cat ./newnewland.grb1 >> pgbfile_${fhr3}_0p25
-  ##0p5 degree
-  # rm -f newland.grb newnewland.grb newnewland.grb1
-  # $WGRIB2 land.grb -set_grib_type same -new_grid_interpolation bilinear -new_grid_winds earth -new_grid $grid0p5 newland.grb
-  # $WGRIB2 newland.grb -set_byte 4 11 218 -grib newnewland.grb
-  # cat ./newnewland.grb >> pgb2file_${fhr3}_0p5
-  #1p0
-  # rm -f newland.grb newnewland.grb newnewland.grb1
-  # $WGRIB2 land.grb -set_grib_type same -new_grid_interpolation bilinear -new_grid_winds earth -new_grid $grid1p0 newland.grb
-  # $WGRIB2 newland.grb -set_byte 4 11 218 -grib newnewland.grb
-  # cat ./newnewland.grb >> pgb2file_${fhr3}_1p0
-  # $CNVGRIB -g21 newnewland.grb newnewland.grb1
-  # cat ./newnewland.grb1 >> pgbfile_${fhr3}_1p0
-  # fi
+  # Move to COM and index the product grib files
+  for grid in "${grids[@]}"; do
+    ${NCP} "pgb2${grp}file_${fhr3}_${grid}" "${COM_ATMOS_GRIB}/${grid}${PREFIX}pgrb2${grp}.${grid}.${fhr3}"
+    ${WGRIB2} -s "pgb2${grp}file_${fhr3}_${grid}" > "${COM_ATMOS_GRIB}/${grid}/${PREFIX}pgrb2${grp}.${grid}.${fhr3}.idx"
```

Comment:
Where is `$COM_ATMOS_GRIB` being defined? Adding the grid here, instead of feeding it into the COM template, violates the paradigm of changes only needing to be in one place (`config.com`).

Also, all of that aside, the first line is missing a `/`.

Reply:
I fixed this.